### PR TITLE
feat(DynamoDB): add DynamoDB service and checks

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

Add DynamoDB service and checks.

### Description

- Add all DynamoDB tests.
-  Add the following checks:

checkName | oldCheck
-- | --
dynamodb_accelerator_cluster_encryption_enabled | extra7165
dynamodb_tables_kms_cmk_encryption_enabled | extra7128
dynamodb_tables_pitr_enabled | extra7151

### CODE REVIEW

File: providers/aws/services/dynamodb/dynamodb_accelerator_cluster_encryption_enabled/dynamodb_accelerator_cluster_encryption_enabled_test.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            assert result[0].status == "FAIL"
+            assert search(
+                "does not have encryption at rest enabled",
+                result[0].status_extended,
+            )
+            assert result[0].resource_id == cluster["ClusterName"]
+            assert result[0].resource_arn == cluster["ClusterArn"]
+
+    @mock_dax
+    def test_dynamodb_table_with_encryption(self):
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_accelerator_cluster_encryption_enabled/dynamodb_accelerator_cluster_encryption_enabled_test.py]: Why table?

File: providers/aws/services/dynamodb/dynamodb_service.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            )
+
+
+@dataclass
+class Table:
+    arn: str
+    name: str
+    encryption_type: str
+    kms_arn: str
+    pitr: bool
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service.py:147]:
```suggestion
    point_in_time_recovery_status: bool
```

File: providers/aws/services/dynamodb/dynamodb_service.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                    TableName=table.name
+                )["ContinuousBackupsDescription"]
+                if "PointInTimeRecoveryDescription" in properties:
+                    if (
+                        properties["PointInTimeRecoveryDescription"][
+                            "PointInTimeRecoveryStatus"
+                        ]
+                        == "ENABLED"
+                    ):
+                        table.pitr = True
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service.py:86]:
```suggestion
                        table.point_in_time_recovery_status = True
```

File: providers/aws/services/dynamodb/dynamodb_service.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        threads = []
+        for regional_client in self.regional_clients.values():
+            threads.append(threading.Thread(target=call, args=(regional_client,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __describe_clusters__(self, regional_client):
+        logger.info("DynamoDB DAX- Describing clusters...")
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service.py]:
```suggestion
        logger.info("DynamoDB DAX - Describing clusters...")
```

File: providers/aws/services/dynamodb/dynamodb_service_test.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            KeySchema=[
+                {"AttributeName": "client", "KeyType": "HASH"},
+                {"AttributeName": "app", "KeyType": "RANGE"},
+            ],
+            BillingMode="PAY_PER_REQUEST",
+        )
+        # DynamoDB client for this test class
+        audit_info = self.set_mocked_audit_info()
+        dynamo = DynamoDB(audit_info)
+        assert len(dynamo.tables) == 2
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service_test.py:99]: Assert the possible values for the `tables` object.

File: providers/aws/services/dynamodb/dynamodb_service_test.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            KeySchema=[
+                {"AttributeName": "client", "KeyType": "HASH"},
+                {"AttributeName": "app", "KeyType": "RANGE"},
+            ],
+            BillingMode="PAY_PER_REQUEST",
+        )["TableDescription"]
+        # DynamoDB client for this test class
+        audit_info = self.set_mocked_audit_info()
+        dynamo = DynamoDB(audit_info)
+        assert len(dynamo.tables) == 1
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service_test.py:126]: Assert the possible values for the `tables` object.

File: providers/aws/services/dynamodb/dynamodb_service_test.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+            BillingMode="PAY_PER_REQUEST",
+        )["TableDescription"]
+        dynamodb_client.update_continuous_backups(
+            TableName="test1",
+            PointInTimeRecoverySpecification={"PointInTimeRecoveryEnabled": True},
+        )
+        # DynamoDB client for this test class
+        audit_info = self.set_mocked_audit_info()
+        dynamo = DynamoDB(audit_info)
+        assert len(dynamo.tables) == 1
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service_test.py:156]: Assert the possible values for the `tables` object.

File: providers/aws/services/dynamodb/dynamodb_service_test.py [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            ClusterName="daxcluster2",
+            NodeType="dax.t3.small",
+            ReplicationFactor=3,
+            IamRoleArn=iam_role_arn,
+            SSESpecification={"Enabled": True},
+        )
+        # DAX client for this test class
+        audit_info = self.set_mocked_audit_info()
+        dax = DAX(audit_info)
+        assert len(dax.clusters) == 2
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service_test.py:186]: Assert the possible values for the `cliusters` object.

File: providers/aws/services/dynamodb/dynamodb_tables_kms_cmk_encryption_enabled/dynamodb_tables_kms_cmk_encryption_enabled.metadata.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
+{
+  "Provider": "aws",
+  "CheckID": "dynamodb_tables_kms_cmk_encryption_enabled",
+  "CheckTitle": "Check if DynamoDB table has encryption at rest enabled using CMK KMS.",
+  "CheckType": ["Software and Configuration Checks/Industry and Regulatory Standards/CIS AWS Foundations Benchmark"],
+  "ServiceName": "dynamodb",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:dynamodb:region:account-id:certificate/resource-id",
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_tables_kms_cmk_encryption_enabled/dynamodb_tables_kms_cmk_encryption_enabled.metadata.json]: Check this arn

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Good job!! Please check my comments

File: providers/aws/services/dynamodb/dynamodb_accelerator_cluster_encryption_enabled/dynamodb_accelerator_cluster_encryption_enabled_test.py (hunk 1 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_accelerator_cluster_encryption_enabled/dynamodb_accelerator_cluster_encryption_enabled_test.py]: Done

File: providers/aws/services/dynamodb/dynamodb_service.py (hunk 2 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service.py:147]: I would leave it as it is.

File: providers/aws/services/dynamodb/dynamodb_service.py (hunk 3 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service.py:86]: I would leave it as it is.

File: providers/aws/services/dynamodb/dynamodb_tables_kms_cmk_encryption_enabled/dynamodb_tables_kms_cmk_encryption_enabled.metadata.json (hunk 9 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_tables_kms_cmk_encryption_enabled/dynamodb_tables_kms_cmk_encryption_enabled.metadata.json]: Done

File: providers/aws/services/dynamodb/dynamodb_service.py (hunk 4 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service.py]: Done

File: providers/aws/services/dynamodb/dynamodb_service_test.py (hunk 5 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service_test.py:99]: Done

File: providers/aws/services/dynamodb/dynamodb_service_test.py (hunk 7 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service_test.py:156]: Done

File: providers/aws/services/dynamodb/dynamodb_service_test.py (hunk 6 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service_test.py:126]: Done

File: providers/aws/services/dynamodb/dynamodb_service_test.py (hunk 8 above)
Comment (MrCloudSec) [reply providers/aws/services/dynamodb/dynamodb_service_test.py:186]: Done

File: providers/aws/services/dynamodb/dynamodb_service_test.py [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+        assert dynamodb.service == "dynamodb"
+
+    # Test Dynamo Client
+    @mock_dynamodb
+    def test_client(self):
+        # Dynamo client for this test class
+        audit_info = self.set_mocked_audit_info()
+        dynamodb = DynamoDB(audit_info)
+        for client in dynamodb.regional_clients.values():
+            assert client.__class__.__name__ == "DynamoDB"
```
Comment (jfagoagas) [inline providers/aws/services/dynamodb/dynamodb_service_test.py]:
```suggestion
        for regional_client in dynamodb.regional_clients.values():
            assert regional_client.__class__.__name__ == "DynamoDB"
```

File: providers/aws/services/dynamodb/dynamodb_service.py (hunk 2 above)
Comment (jfagoagas) [reply providers/aws/services/dynamodb/dynamodb_service.py:147]: As you wish 💋

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**